Assets/Scripts/GameState/PauseManager.cs

```diff
@@ -42,9 +45,23 @@ void Start()
 
     public void PauseGame()
     {
-        IsPaused = !IsPaused;
-        Time.timeScale = IsPaused ? 0 : initialTimeScale;
-        ui.SetActive(IsPaused);
+        if (gameObject.activeSelf)
```

Comment:
If the game object is inactive, no other logic should be performed in this method, and so it's better to invert the `if` statement and return early, which would make the code (slightly) easier to read, due to the function having a reduced cognitive complexity.
```suggestion
        if (!gameObject.activeSelf)
            return;
```